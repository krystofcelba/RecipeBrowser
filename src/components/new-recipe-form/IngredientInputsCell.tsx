import React from 'react';
import { StyleSheet } from 'react-native';
import { Cell } from 'react-native-tableview-simple';
import { Field } from 'redux-form/immutable';

import { View, BoldText } from 'src/components/common';
import AppTheme from 'src/assets/appTheme';
import i18n from 'src/assets/i18n';
import FormTextInput from './FormTextInput';

interface Props {
  index: number;
  name: string;
}

const IngredientInputsCell = ({ index, name }: Props) => (
  <Cell
    cellContentView={
      <View flex={1} flexDirection="row">
        <BoldText>#{index + 1}</BoldText>
        <View style={styles.container}>
          <Field
            style={styles.nameInput}
            placeholder={i18n.t('name')}
            name={`${name}.name`}
            component={FormTextInput}
          />
          <Field
            style={styles.amountInput}
            placeholder={i18n.t('amount')}
            name={`${name}.amount`}
            component={FormTextInput}
          />
          <Field
            style={styles.unitInput}
            placeholder={i18n.t('unit')}
            name={`${name}.unit`}
            component={FormTextInput}
          />
        </View>
      </View>
    }
  />
);

const styles = StyleSheet.create({
  container: {
    paddingHorizontal: AppTheme.defaultPadding,
    width: '100%',
    flexDirection: 'row',
  },
  nameInput: {
    width: '60%',
  },
  amountInput: {
    width: '20%',
    textAlign: 'center',
  },
  unitInput: {
    width: '20%',
    textAlign: 'right',
  },
});

export default IngredientInputsCell;
