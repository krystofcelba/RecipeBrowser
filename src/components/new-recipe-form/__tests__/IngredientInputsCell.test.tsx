import React from 'react';
import { shallow } from 'enzyme';

import IngredientInputsCell from '../IngredientInputsCell';

describe('FormIngredientCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(<IngredientInputsCell name="ingredient0" />);
    expect(wrapper).toMatchSnapshot();
  });
});
