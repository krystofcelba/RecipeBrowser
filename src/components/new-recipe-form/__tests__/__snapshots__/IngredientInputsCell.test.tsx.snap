// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormIngredientCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={
    <glamorous.View
      flex={1}
      flexDirection="row"
    >
      <BoldText>
        #
        NaN
      </BoldText>
      <glamorous.View
        style={
          Object {
            "flexDirection": "row",
            "paddingHorizontal": 10,
            "width": "100%",
          }
        }
      >
        <Field
          component={[Function]}
          name="ingredient0.name"
          placeholder="Name"
          style={
            Object {
              "width": "60%",
            }
          }
        />
        <Field
          component={[Function]}
          name="ingredient0.amount"
          placeholder="Amount"
          style={
            Object {
              "textAlign": "center",
              "width": "20%",
            }
          }
        />
        <Field
          component={[Function]}
          name="ingredient0.unit"
          placeholder="Unit"
          style={
            Object {
              "textAlign": "right",
              "width": "20%",
            }
          }
        />
      </glamorous.View>
    </glamorous.View>
  }
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={false}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title=""
  titleTextColor="#000"
  titleTextStyle={Object {}}
  titleTextStyleDisabled={Object {}}
/>
`;
